controllers/kustomization_controller.go

```diff
@@ -749,43 +761,41 @@ func (r *KustomizationReconciler) apply(kustomization kustomizev1.Kustomization,
 	return changeSet, nil
 }
 
-func (r *KustomizationReconciler) applyWithRetry(kustomization kustomizev1.Kustomization, revision, dirPath string, delay time.Duration) (string, error) {
-	changeSet, err := r.apply(kustomization, dirPath)
+func (r *KustomizationReconciler) applyWithRetry(ctx context.Context, kustomization kustomizev1.Kustomization, revision, dirPath string, delay time.Duration) (string, error) {
+	changeSet, err := r.apply(ctx, kustomization, dirPath)
 	if err != nil {
 		// retry apply due to CRD/CR race
 		if strings.Contains(err.Error(), "could not find the requested resource") ||
 			strings.Contains(err.Error(), "no matches for kind") {
-			r.Log.Info("retrying apply",
-				"error", err.Error(),
-				"kustomization", fmt.Sprintf("%s/%s", kustomization.GetNamespace(), kustomization.GetName()))
+			(logr.FromContext(ctx)).Info("retrying apply", "error", err.Error())
```

Comment:
I am wondering if this is safe to do, as FromContext will return nil if no logger is found in the context? Or is it just fine to assume that the context will always be right.
https://github.com/go-logr/logr/blob/4fa77cb7175ca349567983fd9cc9fd94462f41fa/logr.go#L201

Using FromContextOrDiscard might be a bit better as it returns a discarding logger in the worst case scenario.

Reply:
If the context doesn’t have a logger, then it should panic, note that we set the logger in main.go so the possibility for this is zero.

Reply:
Ok just wanted to check that it was an active choice. I have never been a fan of passing loggers in a context but it seems to be the way a lot of projects are going.

Reply:
The logger is not passed from the context, but the metadata for the logger is collected from the context.